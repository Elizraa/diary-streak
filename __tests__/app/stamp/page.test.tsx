import StampPage from '@/app/stamp/page';
import { render } from '@testing-library/react';
import { act } from 'react';

jest.mock('next/navigation', () => ({
  useRouter: jest.fn(),
}));

it('renders Stamp Page unchanged', async () => {
  let container;

  await act(async () => {
    ({ container } = render(<StampPage />));
  });

  expect(container).toMatchSnapshot();
});
