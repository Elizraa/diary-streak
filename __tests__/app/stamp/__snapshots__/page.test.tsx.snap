// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Stamp Page unchanged 1`] = `
<div>
  <div
    class="min-h-screen bg-gradient-to-b from-gray-950 to-gray-900 flex flex-col items-center justify-center p-4"
  >
    <div
      class="text-card-foreground flex flex-col gap-6 rounded-xl border py-6 w-full max-w-md shadow-lg border-gray-800 bg-gray-900"
      data-slot="card"
    >
      <div
        class="@container/card-header grid auto-rows-min grid-rows-[auto_auto] items-start gap-1.5 px-6 has-data-[slot=card-action]:grid-cols-[1fr_auto] [.border-b]:pb-6 bg-gradient-to-r from-gray-900 to-gray-800 rounded-t-lg border-b border-gray-800 pt-6 -mt-5"
        data-slot="card-header"
      >
        <div
          class="flex items-center justify-center mb-4"
        >
          <svg
            class="lucide lucide-lock h-12 w-12 text-red-400"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              height="11"
              rx="2"
              ry="2"
              width="18"
              x="3"
              y="11"
            />
            <path
              d="M7 11V7a5 5 0 0 1 10 0v4"
            />
          </svg>
        </div>
        <div
          class="leading-none font-semibold text-red-400 text-center"
          data-slot="card-title"
        >
          Access Denied
        </div>
      </div>
      <div
        class="px-6 pt-6 text-center"
        data-slot="card-content"
      >
        <p
          class="text-gray-300 mb-6"
        >
          This page can only be accessed after submitting your daily stamp.
        </p>
      </div>
      <div
        class="items-center px-6 [.border-t]:pt-6 flex justify-center border-t border-gray-800 pt-4"
        data-slot="card-footer"
      >
        <button
          class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive shadow-xs h-9 px-4 py-2 has-[>svg]:px-3 bg-gray-800 hover:bg-gray-700 text-white"
          data-slot="button"
        >
          <svg
            class="lucide lucide-arrow-left mr-2 h-4 w-4"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m12 19-7-7 7-7"
            />
            <path
              d="M19 12H5"
            />
          </svg>
          Return to Home
        </button>
      </div>
    </div>
  </div>
</div>
`;
